script/generate.rb

```diff
@@ -1,7 +1,9 @@
 #!/usr/bin/env brew ruby
 os = ARGV.first
+tap_name = (ARGV.second || "").split("/")
 
 formula_dir = os == "mac" ? "formula" : "formula-linux"
+tap = tap_name.empty? ? CoreTap.instance : Tap.new(*tap_name)
```

Comment:
```suggestion
tap_name = ARGV.second
tap = Tap.new(tap_name.split("/")) if tap_name
tap ||= CoreTap.instance

formula_dir = os == "mac" ? "formula" : "formula-linux"
```

Reply:
@MikeMcQuaid Question: is it preferred for the default core tap to be specified as `CoreTap.instance`? ie, is there some machinery within homebrew that would make `CoreTap.instance` preferable to `Tap.new("homebrew/core")`? If not, we could simplify all this logic by having just a single code path which defaults the tapname argument to `"homebrew/core"`.

Reply:
Actually I just noticed the `Tap.fetch` class method, which seems to do all the normalization, caching and singleton-preference all at once. I think I'll switch to that instead of `Tap.new` ?

Reply:
Sounds good. Happy to use `"homebrew/core"` over `CoreTap.instance`, too.